interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -489,23 +557,26 @@ def _get_cost_node(self, i, N_horizon, xtraj_node, utraj_node, ptraj_node, sl_no
                     ptraj_node[0],
                     sl_node[0],
                     su_node[0],
-                    ocp.get_initial_cost_expression(),
+                    self.get_cost_expression(ocp, ptraj_node[0][dims.np:dims.ny+dims.np], i),
+                    ca.vertcat(ocp.model.p, ptraj_node[0][dims.np:dims.np+dims.ny]),
                     dims.ns_0, cost.zl_0, cost.Zl_0, cost.zu_0, cost.Zu_0)
         elif i < N_horizon:
             return (xtraj_node[i],
                     utraj_node[i],
                     ptraj_node[i],
                     sl_node[i],
                     su_node[i],
-                    ocp.get_path_cost_expression(),
+                    self.get_cost_expression(ocp, ptraj_node[i][dims.np:dims.ny+dims.np], i),
```

Comment:
Here, there is  logic on the order within `p`. Better use the `index*` map instead that you added in this PR.